src/commands/sites/sites-create.js

```diff
@@ -13,6 +13,48 @@ const { link } = require('../link')
 
 const SITE_NAME_SUGGESTION_SUFFIX_LENGTH = 5
 
+const getSiteNameInput = async (name, user, api) => {
+  if (!user) user = await api.getCurrentUser()
+
+  if (!name) {
+    let { slug } = user
+    let suffix = ''
+
+    // If the user doesn't have a slug, we'll compute one. Because `full_name` is not guaranteed to be unique, we
+    // append a short randomly-generated ID to reduce the likelihood of a conflict.
+    if (!slug) {
+      slug = slugify(user.full_name || user.email)
+      suffix = `-${uuidv4().slice(0, SITE_NAME_SUGGESTION_SUFFIX_LENGTH)}`
+    }
+
+    const suggestions = [
+      `super-cool-site-by-${slug}${suffix}`,
+      `the-awesome-${slug}-site${suffix}`,
+      `${slug}-makes-great-sites${suffix}`,
+      `netlify-thinks-${slug}-is-great${suffix}`,
+      `the-great-${slug}-site${suffix}`,
+      `isnt-${slug}-awesome${suffix}`,
+    ]
+    const siteSuggestion = sample(suggestions)
+
+    console.log(
+      `Choose a unique site name (e.g. ${siteSuggestion}.netlify.app) or leave it blank for a random name. You can update the site name later.`,
+    )
+    const { name: nameInput } = await inquirer.prompt([
+      {
+        type: 'input',
+        name: 'name',
+        message: 'Site name (optional):',
+        filter: (val) => (val === '' ? undefined : val),
+        validate: (input) => /^[a-zA-Z\d-]+$/.test(input) || 'Only alphanumeric characters and hyphens are allowed',
+      },
+    ])
+    name = nameInput
+  }
+
+  return name
+}
+
```

Comment:
These changes are only to be able to extract the piece of code that is reused in the `sites-create-template.js` file to avoid duplication, the logic isn't changed